test/langtools/tools/javac/processing/model/util/types/TestInvalidInputs.java

```diff
@@ -0,0 +1,184 @@
+                return (ExecutableType)method.asType();
+            }
+        }
+        throw new RuntimeException("Expected method not found");
+    }
+
+    IntersectionType extractIntersectionType() {
+        var typeElement = elements.getTypeElement("TestInvalidInputs.InvalidInputsHost");
+        for (var method : ElementFilter.methodsIn(typeElement.getEnclosedElements())) {
+            if ("foo9".equals(method.getSimpleName().toString())) {
+                return (IntersectionType) ((TypeVariable)method.getReturnType()).getUpperBound();
+            }
+        }
+        throw new RuntimeException("Expected method not found");
+    }
+
+    /*
+     * Class to host inputs for testing.
+     */
+    class InvalidInputsHost {
+        // Use a method to get an ExecutableType
+        public static String foo7(int arg) {return null;}
+
+        // Type variable with intersection type
+        public static <S extends Number &  Runnable>  S foo9() {return null;}
+    }
+
+    /**
+     * @throws IllegalArgumentException if the given type has no
+     *         unboxing conversion, including for types that are not
+     *         {@linkplain ReferenceType reference types}
+     */
+    void testUnboxedType() {
+        // Only DeclaredType's for wrapper classes should have unboxing converions defined.
+
+        // Reference types are ArrayType, DeclaredType, ErrorType, NullType, TypeVariable
+        // non-reference: ExecutableType, IntersectionType, NoType, PrimitiveType, UnionType, WildcardType
+        var invalidInputs = List.of(objectType, stringType, arrayType,
+                                    executableType, intersectionType,
+                                    noTypeVoid, noTypeNone, noTypePackage, noTypeModule, nullType,
+                                    primitiveType, /*unionType, */ wildcardType);
+
+        for (TypeMirror tm : invalidInputs) {
+            try {
+                PrimitiveType pt = types.unboxedType(tm);
+            } catch(IllegalArgumentException iae) {
+                ; // Expected
+            }
+        }
+        return;
+    }
+
+    /**
+     * @throws IllegalArgumentException if bounds are not valid,
+     * including for types that are not {@linkplain ReferenceType
+     * reference types}
+     */
+    void testGetWildcardType() {
+        // Reference types are ArrayType, DeclaredType, ErrorType, NullType, TypeVariable
+        // non-reference: ExecutableType, IntersectionType, NoType, PrimitiveType, UnionType, WildcardType
+        var invalidInputs = List.of( arrayType,
+                                    executableType, intersectionType,
+                                    noTypeVoid, noTypeNone, noTypePackage, noTypeModule, nullType,
+                                    primitiveType, /*unionType, */ wildcardType);
+
+        for (TypeMirror tm : invalidInputs) {
+            try {
+                WildcardType wc1 = types.getWildcardType(tm,   null);
```

Comment:
Calling getWildcardType for an array type does not throw IAE; is this expected?

Reply:
Will correct in next push; thanks.